.github/workflows/preview-env-deploy.yml

```
@@ -0,0 +1,111 @@
+name: preview-env-deploy
+on:
+  pull_request:
+    types: [labeled, synchronize]
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Very nice, I wasn't aware of this functionality and I may apply it to some other existing workflows. 👍

---

`github.ref` is enough to set a unique group name per branch (and so pull request).

```suggestion
concurrency:
  cancel-in-progress: true
  group: ${{ github.workflow }}-${{ github.ref }}
```

---

> Very nice, I wasn't aware of this functionality and I may apply it to some other existing workflows. 👍

Indeed, it can be set at the workflow or the job level which is really nice also.